File: markdown/src/main/java/com/zzhoujay/markdown/style/LinkSpan.java
package com.zzhoujay.markdown.style;

import android.text.TextPaint;
import android.text.style.URLSpan;

/**
 * Created by zhou on 16-7-2.
 * 链接Span
 */
public class LinkSpan extends URLSpan {

    private int color;

    public LinkSpan(String url, int color) {
        super(url);
        this.color = color;
    }

    @Override
    public void updateDrawState(TextPaint ds) {
        super.updateDrawState(ds);
        ds.setColor(color);
        ds.setUnderlineText(false);
    }

}

File: markdown/src/main/java/com/fastaccess/markdown/spans/LinkSpan.kt
package com.fastaccess.markdown.spans

import android.text.TextPaint
import android.text.style.URLSpan

/**
 * Created by zhou on 16-7-2.
 * 链接Span
 */
class LinkSpan(
        url: String,
        private val color: Int
) : URLSpan(url) {

    override fun updateDrawState(ds: TextPaint) {
        super.updateDrawState(ds)
        ds.color = color
        ds.isUnderlineText = false
    }

}
